core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -1131,14 +1131,43 @@ private[spark] class BlockManager(
   }
 
   /**
+   * Called for pro-active replenishment of blocks lost due to executor failures
+   *
+   * @param blockId blockId being replicate
+   * @param existingReplicas existing block managers that have a replica
+   * @param maxReplicas maximum replicas needed
+   * @return
+   */
+  def replicateBlock(
+    blockId: BlockId,
+    existingReplicas: Set[BlockManagerId],
+    maxReplicas: Int): Unit = {
+    logInfo(s"Pro-actively replicating $blockId")
+    val infoForReplication = blockInfoManager.lockForReading(blockId).map { info =>
```

Comment:
Also, I don't think there's a need to have separate `.map` and `.foreach` calls over the option. Instead, I think it would be clearer to avoid the assignment to the `infoForReplication` variable and just perform all of the work inside of a `.foreach` call on the `Option` with the block info.